src/types.ts

```diff
@@ -0,0 +1,49 @@
+import { DocumentNode } from 'graphql/language/ast';
+import * as Observable from 'zen-observable';
+
+export interface GraphQLRequest {
+  //This string could be an id for a persisted query
+  query?: string | DocumentNode;
+  variables?: object;
```

Comment:
lets make this and context [a `Record` type](https://www.typescriptlang.org/play/index.html#src=let%20x%3A%20Record%3Cstring%2C%20any%3E%0D%0A%0D%0Ax%20%3D%20%7B%20a%3A%201%20%7D%0D%0Ax.b%20%3D%204%0D%0A%0D%0Alet%20y%3A%20Object%0D%0Ay%20%3D%20%7B%20a%3A%201%20%7D%0D%0Ay.b%20%3D%204)